WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -7,7 +7,14 @@ def competition_message_for_user(competition, user)
     if competition.cancelled?
       messages << t('competitions.messages.cancelled')
     elsif registration
-      messages << (registration.accepted? ? t('competitions.messages.tooltip_registered') : t('competitions.messages.tooltip_waiting_list'))
+      # Display a tooltip on a user's bookmarked competition based on their registration status.
+      if registration.accepted?
+        messages << t('competitions.messages.tooltip_registered')
+      elsif registration.deleted?
+        messages << t('competitions.messages.tooltip_deleted')
+      else # If not delted or accepted, assume user is on the waiting list
+        messages << t('competitions.messages.tooltip_waiting_list')
+      end
     end
     visible = competition.showAtAll?
     messages << if competition.confirmed?
```

Comment:
This next step has the `messages <<` factored out to the front. Consistency is good, so I'd suggest doing the same thing on line 9: `messages << if competition.cancelled? ...`. It also makes it clear that this method only adds 2 messages to the array, which is not obvious at a glance with the current code.

Reply:
Thanks Kevin! I've done a slightly more aggressive refactor of the function, that I think achieves that consistency and (at least for me) makes it more readable/followable overall.